infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -270,6 +290,81 @@ private InfrastructureProvisionResult doProvision(Script script, Properties inpu
         }
     }
 
+    /**
+     * Constructs the URL to Kibana dashboard to view logs.
+     *
+     * Makes use of the static dashboard having logs of all test plans. Filters to view logs
+     * of each EC2 instance are derived using placeholders. The default view shows all logs. The shortened URL of the
+     * dashboard is retrieved through Kibana ShortenURL API to have a compact url.
+     *
+     * @param testPlan test-plan to get log url for
+     * @param stackName name of the stack created for the test-plan
+     */
+    private void deriveLogDashboardUrl(TestPlan testPlan, String stackName) {
+        String kibanaEndpoint = ConfigurationContext.getProperty(ConfigurationProperties.KIBANA_ENDPOINT_URL);
+        String dashboardCtxFormat = TestGridConstants.KIBANA_DASHBOARD_STR;
+        String instanceLogFilterFormat = TestGridConstants.KIBANA_FILTER_STR;
+        String instanceLogFilter;
+        String instanceName;
+        StringJoiner filtersStr = new StringJoiner(",");
+
+        // Filter the EC2 instance corresponding to the stack
+        AmazonEC2 amazonEC2 = AmazonEC2ClientBuilder.defaultClient();
+        List<Reservation> reservations = amazonEC2.describeInstances().getReservations()
+                .stream().filter(r -> r.getInstances().get(0).getTags().contains(
```

Comment:
I think we can off-load this filter logic into AWS. See: https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/ec2/model/DescribeInstancesRequest.html#withFilters-com.amazonaws.services.ec2.model.Filter...-

Something like:
```
            DescribeInstancesRequest request = new DescribeInstancesRequest();
            request.withFilters(
                    new Filter("tag:" + AWS_CLOUDFORMATION_STACK_NAME_TAG_KEY).withValues(stackId));
            DescribeInstancesResult result = amazonEC2.describeInstances(request);
```